packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -119,22 +131,45 @@ class DefaultMaterialLocalizations implements MaterialLocalizations {
   ///
   /// [LocalizationsDelegate] implementations typically call the static [load]
   /// function, rather than constructing this class directly.
-  DefaultMaterialLocalizations(this.locale) {
-    assert(locale != null);
-
+  DefaultMaterialLocalizations(this.locale)
+      : assert(locale != null),
+        this._localeName = _computeLocaleName(locale) {
     if (localizations.containsKey(locale.languageCode))
       _nameToValue.addAll(localizations[locale.languageCode]);
     if (localizations.containsKey(_localeName))
       _nameToValue.addAll(localizations[_localeName]);
+
+    if (intl.NumberFormat.localeExists(_localeName)) {
+      _cachedDecimalFormat = new intl.NumberFormat.decimalPattern(_localeName);
+      _cachedTwoDigitZeroPaddedFormat = new intl.NumberFormat('00', _localeName);
+    } else if (intl.NumberFormat.localeExists(locale.languageCode)) {
+      _cachedDecimalFormat = new intl.NumberFormat.decimalPattern(locale.languageCode);
+      _cachedTwoDigitZeroPaddedFormat = new intl.NumberFormat('00', locale.languageCode);
+    } else {
+      _cachedDecimalFormat = new intl.NumberFormat.decimalPattern();
+      _cachedTwoDigitZeroPaddedFormat = new intl.NumberFormat('00');
+    }
   }
 
   /// The locale for which the values of this class's localized resources
   /// have been translated.
   final Locale locale;
 
+  final String _localeName;
+
   final Map<String, String> _nameToValue = <String, String>{};
 
-  String get _localeName {
+  /// Formats numbers using variable length format with no zero padding.
+  ///
+  /// See also [_cachedTwoDigitZeroPaddedFormat].
+  intl.NumberFormat _cachedDecimalFormat;
+
+  /// Formats numbers as two-digits.
+  ///
+  /// If the number is less than 10, zero pads it.
+  intl.NumberFormat _cachedTwoDigitZeroPaddedFormat;
```

Comment:
I would remove the _cached prefix, since it's eagerly computed and never changed.